src/gen_smtp_client.erl

```diff
@@ -121,7 +123,46 @@ send_it_nonblock(Email, Options, Callback) ->
 			{ok, Receipt}
 	end.
 
--spec send_it(Email :: {string() | binary(), [string() | binary(), ...], string() | binary() | function()}, Options :: list()) -> binary() | {'error', any(), any()}.
+-opaque smtp_client_socket() :: {socket:socket(), list(), list()}.
+-spec open(Options :: list()) -> SocketDescriptor :: smtp_client_socket() | {error, any()}.
+%% @doc Open a SMTP client socket with the provided options
+%% Once the socket has been opened, you can use it with deliver/2.
+open(Options) ->
+	NewOptions = lists:ukeymerge(1, lists:sort(Options),
+															 lists:sort(?DEFAULT_OPTIONS)),
+	case check_options(NewOptions) of
+		ok ->
+			RelayDomain = proplists:get_value(relay, NewOptions),
+			MXRecords = case proplists:get_value(no_mx_lookups, NewOptions) of
+										true ->
+											[];
+										_ ->
+											smtp_util:mxlookup(RelayDomain)
+									end,
+			%io:format("MX records for ~s are ~p~n", [RelayDomain, MXRecords]),
+			Hosts = case MXRecords of
+								[] ->
+									[{0, RelayDomain}]; % maybe we're supposed to relay to a host directly
+								_ ->
+									MXRecords
+							end,
+			try_smtp_sessions(Hosts, NewOptions, []);
+		{error, Reason} ->
+			{error, bad_option, Reason}
+	end.
+
+-spec deliver(Socket :: smtp_client_socket(), Email :: email()) -> Receipt :: binary() | {error, any()}.
```

Comment:
Shall we make the return value `{ok, binary()}`, for easier pattern matching?

The same for the `open/1` function: `{ok, smtp_client_socket()}`